core/rfb.js

```diff
@@ -1543,6 +1872,90 @@
             sock.flush();
         },
 
+        atenKeyEvent: function (sock, keysym, down) {
+            var ks = XK2HID[keysym];
```

Comment:
I took a stab at this and found a further complication: what the browser supplies[1] is *not* a USB HID usage code ("scan-code")[2].

[1] https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
[2] http://www.usb.org/developers/hidpage/Hut1_12v2.pdf, ch. 10 (page 53)